core/pkg/launch/job_builder.go

```diff
@@ -153,11 +154,14 @@ type JobBuilder struct {
 
 	PartialJobSource *PartialJobSource
 
-	Disable         bool
-	settings        *service.Settings
-	RunCodeArtifact *ArtifactInfoForJob
-	aliases         []string
-	isNotebookRun   bool
+	Disable               bool
+	settings              *service.Settings
+	RunCodeArtifact       *ArtifactInfoForJob
+	aliases               []string
+	isNotebookRun         bool
+	runConfig             map[string]interface{}
```

Comment:
Ah, this is very useful. I will refactor this PR to use `RunConfig`

Reply:
The launch package is imported in `server` so this creates a circular import. I think the way around this would be to move `job_builder.go` from `launch` to `server`.

Reply:
I think the right approach here might be to move `config.go` into its own package, probably `internal/runconfig/runconfig.go`. I can send out a PR for that.

Reply:
It might make more sense to move the RunConfig code into its own package assuming it does not have dependancies on other aspects of the `server` package

Reply:
https://github.com/wandb/wandb/pull/7112

Reply:
yep, @timoffex and I were actually discussing it today, that the server pkg has too much things going on in it... and it would be actually nice to reduce its size. So i tend to agree with moving config into its own pkg. 

@timoffex maybe follow up would be to do the same for summary ...

Reply:
We will want to use this library to represent and filter other than the `wandb.config` as well.